.gitlab-ci.yml

```diff
@@ -325,6 +325,23 @@ pkg:nix:deploy:
     - master
     - /^v.*\..*$/
 
+pkg:nix:deploy:channel:
+  stage: deploy
+  environment:
+    name: cachix
+    url: https://coq.cachix.org
+  dependencies:
+    - pkg:nix:deploy
+  before_script:
+    - which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )
+    - eval $(ssh-agent -s)
+    - echo "$GITHUB_DEPLOY_KEY" | tr -d '\r' | ssh-add - > /dev/null
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+    - ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
```

Comment:
Maybe, most of this `before_script` (as well as the `stage`  property) could me moved to a template and shared with the other deployment job.

Reply:
Yes but I don't know how to do this properly: GitLab templates only allow sharing the full `before_script` or nothing in it.

Reply:
There is only one line that is specific to each job (the one adding the ssh key). Said line might be move to the `script` part…